pkg/route/api/validation/validation.go

```diff
@@ -76,6 +79,64 @@ func ValidateRouteStatusUpdate(route *routeapi.Route, older *routeapi.Route) fie
 	return allErrs
 }
 
+// ValidateRouteTLSConfig checks if the TLS config (certificates and keys) for a route is valid.
+func ValidateRouteTLSConfig(route *routeapi.Route) field.ErrorList {
```

Comment:
This is actually called from outside - i will call it extended route validation or something to that effect for clarity.